.github/workflows/auto-update-upgrade.yml

```diff
@@ -0,0 +1,27 @@
+name: On Release Auto Upgrade
```

Comment:
This is awesome. Can you please add docs/comments directly in this file explaining what this CI job does? Excited to try it next upgrade!

Reply:
yes absolutely will!